### ISSUE TITLE
pvtu (unstructured)

### ISSUE THREAD

**Issue by AliE89** [szaghi/VTKFortran#32]
Ciao Stefano,

Do you have an example to write pvtu? 

I can only find pvts in the examples and the source file seems to include pvts only. 

Thanks!

**Comment by AliE89** [comment]
Ok, sorry I have misunderstood the usage. It works for pvtu. Here what I have done, it can be usefull to other users as well:

			ierr = pvtk_field_file%initialize(filename='postProc/VTKfields/Parallel/mesh.pvtu', &
					                          mesh_topology='PUnstructuredGrid', mesh_kind="Float64")

			ierr = pvtk_field_file%xml_writer%write_dataarray(location='cell', action='open')

			!error = pvtk_field_file%xml_writer%write_parallel_dataarray(data_name='int32_scalar', &
			!		data_type='Int32', &
			!		number_of_components=1)

			ierr = pvtk_field_file%xml_writer%write_dataarray(location='cell', action='close')

			ierr = pvtk_field_file%xml_writer%write_parallel_geo(source="./file1.vtu")
			ierr = pvtk_field_file%xml_writer%write_parallel_geo(source="./file2.vtu")

			ierr = pvtk_field_file%finalize()

**Comment by szaghi** [comment]
@AliE89 

I am sorry for my delay.

Have you solved it? I have to improve examples and docs, my bad.

**Comment by AliE89** [comment]
Hi Stefano,

Yes it was really easy doing that, i found an example in the doc in the end
:)

I am still a bit puzzled on polyheadral cells, any suggestion?

**Comment by szaghi** [comment]
@AliE89 

> I am still a bit puzzled on polyheadral cells, any suggestion?

Indeed, nope because I have never used poly output. I'll try to figure out how complicated is to support it (I guess not so complicated), but I cannot plan a timeline right now, I am really too much busy with my official work.

**Comment by AliE89** [comment]
Hi,

Hope Stefano does not mind if I reply to your question.

You cannot write a pvtu file directly, but pvtu is just a plain file where
you basically link toghether N vtu files.

So if your code is parallel , you let each partition write a vtu file with
vtk fortran then your master node just writes down the pvtu.

You can see an example at:
https://gitlab.com/alie89/flubio-code-fvm/-/blob/main/src/pre_post_proc/parallelPost/formats/vtkFormat.f03?ref_type=heads

I can provide you an example file tomorrow.

Cheers

Il Gio 19 Giu 2025, 15:48 lj-cug ***@***.***> ha scritto:

**Comment by szaghi** [comment]
Hi @lj-cug I am sorry, but I am currently busy. I'll try to provide an example next week, but I cannot promise.

Thank you @AliE89 for your help, it is really appreciated 👍 

@lj-cug, you can save ghost cells as well as the inner-domain cells. I do this often: simply pass the whole domain to the grid (topology and fields routine) instead of just the inner cells.

Stefano
